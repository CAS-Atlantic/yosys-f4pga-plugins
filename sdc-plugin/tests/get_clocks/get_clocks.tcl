yosys -import
plugin -i sdc
plugin -i design_introspection
# Import the commands from the plugins to the tcl interpreter
yosys -import

read_verilog $::env(DESIGN_TOP).v
read_verilog -specify -lib -D_EXPLICIT_CARRY +/xilinx/cells_sim.v
read_verilog -lib +/xilinx/cells_xtra.v
hierarchy -check -auto-top
# Start flow after library reading
synth_xilinx -flatten -abc9 -nosrl -nodsp -iopad -run prepare:check
#synth_xilinx

# Read the design's timing constraints
read_sdc $::env(DESIGN_TOP).input.sdc

# Propagate the clocks
propagate_clocks

# Write the clocks to file
set fh [open $::env(DESIGN_TOP).txt w]

puts $fh [get_clocks]

puts $fh [get_clocks -include_generated_clocks]

puts $fh [get_clocks -include_generated_clocks clk2]

puts $fh [get_clocks -of [get_nets clk_int_1 clk1] -include_generated_clocks clk_int_1]

puts $fh [get_clocks -of [get_nets]]

puts $fh [get_clocks -of [concat [get_nets clk2] [get_nets clk_int_1 clk]]]

close $fh
