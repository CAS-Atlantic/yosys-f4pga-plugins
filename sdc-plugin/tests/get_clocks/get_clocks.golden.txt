{$auto$clkbufmap.cc:262:execute$1800} {$auto$clkbufmap.cc:262:execute$1802} clk clk2 clk_int_1 middle_inst_1.clk_int middle_inst_4.clk
{$auto$clkbufmap.cc:262:execute$1800} {$auto$clkbufmap.cc:262:execute$1802} clk clk2 clk_int_1 middle_inst_1.clk_int middle_inst_4.clk
clk2
clk_int_1
clk clk2 clk_int_1 middle_inst_1.clk_int middle_inst_4.clk
clk clk2 clk_int_1
